### ISSUE TITLE
esbuild typescript support?

### ISSUE THREAD

**Issue by marcustut** [exhibitionist-digital/ultra#14]
```tsx
import React, { lazy, Suspense } from "react";
import { Helmet } from "helmet";
import { Route } from "wouter";
import { SWRConfig } from "swr";
import ultraCache from "ultra/cache";

const Index = lazy(() => import("./index.tsx"));

const Ultra = ({ cache }: { cache: Map<unknown, unknown> }) => {
  return (
    <SWRConfig value={{ provider: () => ultraCache(cache) }}>
      <Helmet>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta charSet="UTF-8" />
        <link rel="stylesheet" href="/style.css?react18" />
        <title>Ultra: React 18 Streaming SSR</title>
        <link
          rel="icon"
          type="image/svg+xml"
          href="https://ultrajs.dev/logo.svg"
        />
      </Helmet>
      <Suspense fallback={null}>
        <Route path="/" component={Index} />
      </Suspense>
    </SWRConfig>
  );
};

export default Ultra;
```

When running the above code with `make dev`, which essentially is `mode=dev deno run --no-check --allow-net --allow-read --allow-env --allow-run --allow-write --import-map=importmap.json --unstable server.ts`

it throws the following error:
```
Error: Transform failed with 1 error:
<stdin>:9:24: error: Expected ")" but found ":"
    at failureErrorWithLog (https://deno.land/x/esbuild@v0.12.24/mod.js:1414:15)
    at https://deno.land/x/esbuild@v0.12.24/mod.js:1225:29
    at https://deno.land/x/esbuild@v0.12.24/mod.js:570:9
    at handleIncomingPacket (https://deno.land/x/esbuild@v0.12.24/mod.js:667:9)
    at readFromStdout (https://deno.land/x/esbuild@v0.12.24/mod.js:537:7)
    at https://deno.land/x/esbuild@v0.12.24/mod.js:1684:11
Download http://localhost:3000/app.js?ts=1632075424632
^Cmake: *** [Makefile:2: dev] Interrupt
```

If I remove the type for `cache` which is changing from `const Ultra = ({ cache }: { cache: Map<unknown, unknown> }) => {` to `const Ultra = ({ cache }) => {` then everything works fine. Therefore, I am suspecting that esbuild does not recognize this as a TypeScript file.

**Comment by mashaal** [comment]
Thanks @marcustut - this will be fixed in the next release, it should go out later today or tomorrow. I will add types for the SWR cache as well.

https://github.com/exhibitionist-digital/ultra/commit/912356bb589603b3a90699eaf0b64646fd24f36e

**Comment by mashaal** [comment]
This should be fixed now in v0.6: https://github.com/exhibitionist-digital/ultra/releases/tag/v0.6

I also updated the ultra-website example to use TS: https://github.com/exhibitionist-digital/ultra/blob/master/examples/ultra-website/src/app.tsx
